Objects/typeobject.c

```diff
@@ -3384,6 +3390,12 @@ PyType_FromModuleAndSpec(PyObject *module, PyType_Spec *spec, PyObject *bases)
     char *res_start;
     short slot_offset, subslot_offset;
 
+    if (meta->tp_new != PyType_Type.tp_new) {
+        PyErr_SetString(PyExc_SystemError,
```

Comment:
I'm not sure if SystemError is the most appropriate here; perhaps TypeError?

Reply:
FTR, `TypeError` is raised from within `PyType_FromModuleAndSpec` if `base` does not have the `Py_TPFLAGS_BASETYPE` flag.

Reply:
I agree that `TypeError` makes more sense and switched to that.